docs/reference/search/search-your-data/retrievers_examples.asciidoc

```diff
@@ -0,0 +1,327 @@
+retriever. This retriever operates on top of two other retrievers: a `knn` retriever and a
+`standard` retriever. Our query structure would look like this:
+
+[source,js]
+----
+GET /retrievers_example/_search
+{
+   "retriever":{
+       "rrf": {
+           "retrievers":[
+               {
+                   "standard":{
+                       "query":{
+                           "semantic":{
+                               "field": "inference_field",
+                               "query": "Can I use generative AI to identify user intent and improve search relevance?"
+                           }
+                       }
+                   }
+               },
+               {
+                   "knn": {
+                       "field": "vector",
+                       "query_vector": [0.23, 0.67, 0.89],
+                       "k": 3,
+                       "num_candidates": 5
+                   }
+               }
+           ],
+           "rank_window_size": 10,
+           "rank_constant": 1
+       }
+   },
+   "_source": ["text", "topic"]
+}
+----
+//NOTCONSOLE
+end::basic-rrf-retriever-with-semantic-query[]
+
+tag::rrf-retriever-with-collapse[]
+[discrete]
+=== Grouping results by year with `collapse`
+
+In our result set, we have many documents with the same `year` value. We can clean this
+up using the `collapse` parameter with our retriever. This enables grouping results by
+any field and returns only the highest-scoring document from each group. In this example
+we'll group by year.
+on a given value and pick just the top for each sub-group!
+
+[source,js]
+----
+GET /retrievers_example/_search
+{
+   "retriever":{
+       "rrf": {
+           "retrievers":[
+               {
+                   "standard":{
+                       "query":{
+                           "semantic":{
+                               "field": "inference_field",
+                               "query": "Can I use generative AI to identify user intent and improve search relevance?"
+                           }
+                       }
+                   }
+               },
+               {
+                   "knn": {
+                       "field": "vector",
+                       "query_vector": [0.23, 0.67, 0.89],
+                       "k": 3,
+                       "num_candidates": 5
+                   }
+               }
+           ],
+           "rank_window_size": 10,
+           "rank_constant": 1
+       }
+   },
+   "collapse": {
+       "field": "year",
+       "inner_hits": {
+           "name": "topic related documents",
+           "_source": ["text", "year"]
+       }
+   },
+   "_source": ["text", "topic"]
+}
+----
+//NOTCONSOLE
+end::rrf-retriever-with-collapse[]
+
+tag::rrf-on-top-of-semantic-reranker[]
+[discrete]
+=== RRF with semantic reranker
+
+For this example, we'll replace our semantic query with the `my-awesome-rerank-model`
```

Comment:
I see you have semantic reranker on top of the previous two retrievers in the following example. For me it would make more sense to have it as the first example:

- It makes more sense IMO as we're reranking as part of a final stage on top
- We build directly on top of the previous example
- We build real-world retrievers usage first by addressing hybrid search and reranking first

Then you can keep exploring the full composability in the following examples.

Reply:
So would you suggest to remove the `RRF with semantic reranker` example, or change the order to something like the following? 
* `Rerank results of RRF retriever` (building on previous example)
* `RRF with semantic reranker`.  (swapping the rrf with the semantic reranker retrievers)
* `Chaining multiple semantic rerankers`

Reply:
Updated order to: 
```
include::retrievers_examples.asciidoc[tag=basic-rrf-retriever-with-semantic-query]
include::retrievers_examples.asciidoc[tag=rrf-retriever-with-collapse]
include::retrievers_examples.asciidoc[tag=text-similarity-reranker-on-top-of-rrf]
include::retrievers_examples.asciidoc[tag=rrf-on-top-of-semantic-reranker]
include::retrievers_examples.asciidoc[tag=chaining-text-similarity-reranker-retrievers]
include::retrievers_examples.asciidoc[tag=rrf-retriever-with-aggs]
```